exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -695,6 +698,33 @@ public void runQuery(QueryType type, List<PlanFragment> planFragments, UserResul
   }
 
   /**
+   * Get server properties that represent the list of server session options.
+   *
+   * @return server properties for the server session options.
+   */
+  public ServerProperties getOptions() throws RpcException {
```

Comment:
I don't think this is the right interface to expose to the user as this is too generic and introduce too much indirect coupling (clients starting dependending on specific options). To be clear, I'm the one to blame here. When we discussed adding metadata methods to the JDBC/ODBC client, one of the things discussed was a server info metadata, to return things like quoting or some other properties (equivalent to JDBC DatabaseMetadata object, or the C++ connector Metadata class: https://github.com/apache/drill/blob/master/contrib/native/client/src/include/drill/drillClient.hpp#L712), but I didn't free some time to add the missing RPC call.

This is probably what we should use here, and hopefully I can probably add it before end of week.